AnkiDroid/src/main/java/com/ichi2/utils/JSONObject.java

```diff
@@ -0,0 +1,289 @@
+package com.ichi2.utils;
+
+/**
+ * Each method similar to the methods in JSONObjects. Name changed to add a ,
+ * and it throws JSONException instead of JSONException.
+ * Furthermore, it returns JSONObject and JSONArray
+ *
+ */
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class JSONObject extends org.json.JSONObject {
+
+    public static final Object NULL = JSONObject.NULL;
+
+    public JSONObject() {
+        super();
+    }
+
+    public JSONObject(Map copyFrom) {
+        super(copyFrom);
+    }
+
+    // original code from https://github.com/stleary/JSON-java/blob/master/JSONObject.java
+    // super() must be first instruction, thus it can't be in a try, and the exception can't be catched
```

Comment:
this comment then explains it - java doesn't allow it to be in a try. That is unfortunate. while I dislike duplication, if the code must be duplicated then I'd like to see this comment duplicated as well or I will wonder every time. Also, tiny grammar change because everything in English is irregular

```suggestion
    // super() must be first instruction, thus it can't be in a try, and the exception can't be caught
```